packages/query-core/src/query.ts

```diff
@@ -473,8 +475,8 @@ export class Query<
         this.isFetchingOptimistic = false
       },
       onError,
-      onFail: () => {
-        this.dispatch({ type: 'failed' })
+      onFail: (_, error) => {
+        this.dispatch({ type: 'failed', error })
```

Comment:
there seems to be no consumer that actually needs the `failureCount` that is passed here as the first argument to `onFail`, so maybe we can just remove that? The `failureCount` is just incremented internally ...

Or, we use the `failureCount` here and pass it to our internal reducer instead of incrementing it there. I just think the middle ground we have now is kinda weird because we increment in two places:

https://github.com/TanStack/query/blob/ca767771f3720ba9abb541871c22ea106172e9ab/packages/query-core/src/retryer.ts#L176-L179

Reply:
Since `failureCount` is used within `retryer.ts` at multiple points it might make more sense to use this value in the `reducer` as well. I will change it accordingly